File: Assignment4/Q4.c
/////////////////////////////////////////////////////////////////////////////////
//
//      Include required header files
//
/////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////
//
//      Function name:  SumNonFact
//      Input:          integer
//      Output:         integer
//      Description:    Calculates sum of non factors of given number
//      Date:           24/05/2026
//      Author:         Pratik Subhash Mahale
//
/////////////////////////////////////////////////////////////////////////////////
int SumNonFact(
                    int iNo         // Input number
              )
{
    int i = 0;                      // Counter variable
    int iSum = 0;                   // Variable to store summation

    // Convert negative number into positive
    if(iNo < 0)
    {
        iNo = -iNo;
    }

    // Loop to find non factors
    for(i = 1; i < iNo; i++)
    {
        // Check whether i is non factor
        if(iNo % i != 0)
        {
            iSum = iSum + i;        // Add non factors
        }
    }

    return iSum;                    // Return summation
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Application to calculate sum of non factors
//
/////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue = 0;                 // Variable to store input number
    int iRet = 0;                   // Variable to store return value

    printf("Enter number\n");

    scanf("%d", &iValue);

    // Function call
    iRet = SumNonFact(iValue);

    // Display result
    printf("%d", iRet);

    return 0;                       // Successful termination
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Input:      12
//      Output:     50
//
/////////////////////////////////////////////////////////////////////////////////

File: Assignment21/Q4.java
///////////////////////////////////////////////////////////////////
//
//      File Name : Q4.java
//      Project   : Logic Building Assignment 21
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : CountFactorLogic
//      Method Name: countFactors
//      Input      : Integer
//      Output     : Void
//      Description: Count the total number of
//                   factors of a given number
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class CountFactorLogic
{
    void countFactors(int num)
    {
        int i = 0;          // Loop counter
        int iCount = 0;     // Store factor count

        // Convert negative number to positive
        if(num < 0)
        {
            num = -num;
        }

        // Count factors
        for(i = 1; i <= num; i++)
        {
            if(num % i == 0)
            {
                iCount++;
            }
        }

        // Display result
        System.out.println("Total factors are : " + iCount);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q4
{
    public static void main(String args[])
    {
        // Create object of CountFactorLogic class
        CountFactorLogic obj = new CountFactorLogic();

        // Function call
        obj.countFactors(20);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 20
//
//      Output :
//      Total factors are : 6
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Factors of 20 : 1 2 4 5 10 20
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Space Complexity : O(1)
//
///////////////////////////////////////////////////////////////////

